packages/grafana-ui/src/components/uPlot/utils.ts

```diff
@@ -152,6 +149,196 @@ export function collectStackingGroups(f: Field, groups: Map<string, number[]>, s
   }
 }
 
+export interface StackingGroup {
+  series: number[];
+  dir: number;
+}
+
+// generates bands between adjacent group series
+export function getStackingBands(group: StackingGroup) {
+  let bands: uPlot.Band[] = [];
+  let { series, dir } = group;
+  let lastIdx = series.length - 1;
+
+  let rSeries = series.slice().reverse();
+
+  rSeries.forEach((si, i) => {
+    if (i !== lastIdx) {
+      let nextIdx = rSeries[i + 1];
+      bands.push({
+        series: [si, nextIdx],
+        // fill direction is inverted from stack direction
+        dir: -1 * dir,
+      });
+    }
+  });
+
+  return bands;
+}
+
+// expects an AlignedFrame
+export function getStackingGroups(frame: DataFrame) {
```

Comment:
is it possible to write tests for this function? Would help

Reply:
the updated tests for `preparePlotData2()` depend on this to work properly, so it's covered by association:

```js
expect(preparePlotData2(df, getStackingGroups(df)))
```

do you still prefer to make these more unit-y?

Reply:
added some tests